File: user/src/controllers/suggestion.controller.js
//Editing now
import suggestionService from '../services/suggestion.service.js';


const getAllSuggestions = async (req, res,) => {
    console.log('suggestion controller');
    try {
        const suggestions = await suggestionService.getAllSuggestions();
        res.status(200).json(suggestions);
    } catch (error) {
        res.status(500).json({ message: error.message });
    }
};

const getSuggestionById = async (req, res) => {
    console.log('suggestion controller');
    const suggestionId = req.params.id;
    try {
        const suggestion = await suggestionService.getSuggestionById(suggestionId);
        if (!suggestion) {
            return res.status(404).json({ message: 'Suggestion not found' });
        }
        // req.user = user; 
        res.status(200).json(suggestion);
    } catch (error) {
        res.status(500).json({ message: error.message });
    }
};

const getSuggestionByCreator = async (req, res) => {
    console.log('suggestion controller');
    const creatorId = req.params.id;
    try {
        const suggestion = await suggestionService.getSuggestionByCreator(creatorId);
        if (!suggestion) {
            return res.status(404).json({ message: 'Suggestion not found' });
        }
        // req.user = user; 
        res.status(200).json(suggestion);
    } catch (error) {
        res.status(500).json({ message: error.message });
    }
};

const createSuggestion = async (req, res) => {
    console.log('suggestion controller');
    console.log(req.body);
    try {
        const suggestion = await suggestionService.createSuggestion(req.body);
        res.status(201).json(suggestion);
    } catch (error) {
        console.log(error.message)
        res.status(400).json({ message: error.message });
    }
};

const updateSuggestion = async (req, res) => {
    console.log('suggestion controller');
    console.log(req.body);
    console.log(req.params.id); 
    const suggestionId = req.params.id;
    const updatedData = req.body;
    try {
        const updatedSuggestion = await suggestionService.updateSuggestion(suggestionId, updatedData);
        res.status(201).json(updatedSuggestion);
    } catch (error) {
        console.log(error.message)
        res.status(400).json({ message: error.message });
    }
};



export {getAllSuggestions,getSuggestionById,getSuggestionByCreator,createSuggestion,updateSuggestion};

File: Admin-Spring/src/main/java/com/ibm/springboot/cityManagement/controller/SuggestionController.java
//package com.ibm.springboot.cityManagement.controller;
//
//import org.springframework.beans.factory.annotation.Autowired;
//import org.springframework.http.ResponseEntity;
//import org.springframework.web.bind.annotation.*;
//
//import com.ibm.springboot.cityManagement.model.Suggestion;
//import com.ibm.springboot.cityManagement.service.SuggestionService;
//
//@RestController
//@RequestMapping("/suggestions")
//public class SuggestionController {
//
//    @Autowired
//    private SuggestionService suggestionService;
//
//    @GetMapping
//    public ResponseEntity<?> getAllSuggestions() {
//        try {
//            return ResponseEntity.ok(suggestionService.getAllSuggestions());
//        } catch (Exception e) {
//            return ResponseEntity.status(500).body(e.getMessage());
//        }
//    }
//
//    @GetMapping("/suggestion-by-id/{id}")
//    public ResponseEntity<?> getSuggestionById(@PathVariable String id) {
//        try {
//            Suggestion suggestion = suggestionService.getSuggestionById(id);
//            if (suggestion == null) {
//                return ResponseEntity.status(404).body("Suggestion not found");
//            }
//            return ResponseEntity.ok(suggestion);
//        } catch (Exception e) {
//            return ResponseEntity.status(500).body(e.getMessage());
//        }
//    }
//
//    @GetMapping("/suggestion-by-creatorId/{creatorId}")
//    public ResponseEntity<?> getSuggestionByCreator(@PathVariable String creatorId) {
//        try {
//            Suggestion suggestion = suggestionService.getSuggestionByCreator(creatorId);
//            if (suggestion == null) {
//                return ResponseEntity.status(404).body("Suggestion not found");
//            }
//            return ResponseEntity.ok(suggestion);
//        } catch (Exception e) {
//            return ResponseEntity.status(500).body(e.getMessage());
//        }
//    }
//
//    @PostMapping
//    public ResponseEntity<?> createSuggestion(@RequestBody Suggestion suggestion) {
//        try {
//            Suggestion newSuggestion = suggestionService.createSuggestion(suggestion);
//            return ResponseEntity.status(201).body(newSuggestion);
//        } catch (Exception e) {
//            return ResponseEntity.status(400).body(e.getMessage());
//        }
//    }
//
//    @PutMapping("/{id}")
//    public ResponseEntity<?> updateSuggestion(@PathVariable String id, @RequestBody Suggestion updatedSuggestion) {
//        try {
//            Suggestion suggestion = suggestionService.updateSuggestion(id, updatedSuggestion);
//            if (suggestion == null) {
//                return ResponseEntity.status(404).body("Suggestion not found");
//            }
//            return ResponseEntity.ok(suggestion);
//        } catch (Exception e) {
//            return ResponseEntity.status(400).body(e.getMessage());
//        }
//    }
//}
